python/lsst/ctrl/mpexec/mock_task.py

```diff
@@ -85,7 +86,12 @@ def _get(self, ref: Optional[Union[DeferredDatasetRef, DatasetRef]]) -> Any:
 
         try:
             mockDatasetType = self.butler.registry.getDatasetType(mockDatasetTypeName)
-            ref = DatasetRef(mockDatasetType, ref.dataId)
+            # Is the ref needed here? Can those parameters be passed directly
+            # to the get() call or is this ref needed in the KeyError except
+            # catch?
```

Comment:
Hmm, I think it should work if dateset type and dataId are passed to butler.get(), but ref is needed in `except` clause below for `super()._get()`. Also `KeyError` should probably be `LookupError`. I have not tested mock tasks for some time, there is a chance that all of this is somewhat broken.

Reply:
Right, so you create a ref in the try block specifically so it will overwrite the main ref and then be used in the except clause. Doesn't that mean the except clause is using mockDatasetType even though the comment after the `_get(ref)` call is implying it was a real dataset.

Reply:
I finally understand. The look up failure happens in the mockDatasetType definition itself. The change of the value of the `ref` variable is irrelevant. I was assuming the error came from the `butler.get`. That means the `ref` in the try block is not needed at all.